src/formatters/AnsiConsoleFormatter.js

```diff
@@ -69,6 +69,58 @@ class AnsiConsoleFormatter extends Formatter {
         }
         return output;
     }
+
+    formatOutput(options){
+        let prjName, totalTime, fileStats, score, resultStats, results, errorsOnly;
+
+        if (options) {
+            prjName = options.name;
+            totalTime = options.time;
+            fileStats = options.fileStats;
+            resultStats = options.resultStats;
+            results = options.results;
+            score = options.score;
+            errorsOnly = options.errorOnly;
+        }
+
+        if (results) {
+            results.forEach(result => {
+                const str = this.format(result);
+                if (str) {
+                    if (result.severity === "error") {
+                        logger.error(str);
```

Comment:
formatOutput is supposed to return a string of the fully formatted file and not call the logger to output things on the fly

Reply:
then I'm a bit confused. The `AnsiConsoleFormatter` need to use logger.error, logger.warn, and logger.info depending on the severity of the result. when should I call them? Within the format()?

Reply:
I update it. Please check them again

Reply:
Here is an way that I tested with log4js. 
I wanted to see if any log from other files will write to log file,
In conclusion, It doesn't work that I expected. ;( Do I did something wrong?

``` 
// src/index.js
if (options.opt.output) {
    log4js.configure({
        appenders: {
            default: { 
                "type": "stdout",
                "layout": {"type": "pattern", "pattern": "%[%p%]: %m"}
            },
            output: {
              type: "file", filename: "temp.log",},
          },
          categories: {
            default: { appenders: ["default", "output"], level: "info" },
            outputFile: { appenders: ["output"], level: "info" },
          },
    });
}
```
```
// src/Project.js
const logger = log4js.getLogger("outputFile");
logger.debug("Got cheese.");
logger.info("Cheese is Comté.");
logger.warn("Cheese is quite smelly.");
logger.error("Cheese is too ripe!");
```
```
// temp.log
It is only log message from src/index.js
[2024-03-27T15:29:16.782] [INFO] ilib-lint.root - ilib-lint - Copyright (c) 2022-2024 JEDLsoft, All rights reserved.
```

Reply:
Also, the file name of the logger could not be set externally.
e.g) not working: 
```
let resultFileName = "lint-result.txt"
.....
appenders: {
   output: {type: "file", filename: resultFileName,},},
....
```

Reply:
I like the test messages, haha.

Did you remove the previous log4js.configure() call on line 35 of index.js first? I think you can only configure it once...

Reply:
the messages are from the npm log4js package example ;)

Yes I also tested your suggested way by removing line 35. The problem is that it only works in the index.js file. I expected it to affect other files as well, but it didn't. I also tried to create log4js.configure in Project.js in the same way, but nothing is written to the file, just an empty file with the name I specified in configure. hmm